pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -394,44 +387,40 @@ public synchronized PinotResourceManagerResponse addInstance(@Nonnull Instance i
   }
 
   /**
-   * Get all Pinot realtime table names.
+   * Get all offline table names.
    *
-   * @return List of Pinot realtime table names
+   * @return List of offline table names
    */
-  @Nonnull
-  public List<String> getAllRealtimeTables() {
-    List<String> resourceNames = getAllResources();
-    Iterator<String> iterator = resourceNames.iterator();
-    while (iterator.hasNext()) {
-      if (!TableNameBuilder.REALTIME.tableHasTypeSuffix(iterator.next())) {
-        iterator.remove();
+  public List<String> getAllOfflineTables() {
+    List<String> offlineTableNames = new ArrayList<>();
+    for (String resourceName : getAllResources()) {
+      if (TableNameBuilder.OFFLINE.tableHasTypeSuffix(resourceName)) {
+        offlineTableNames.add(resourceName);
       }
     }
-    return resourceNames;
+    return offlineTableNames;
   }
 
   /**
-   * Get all Pinot offline table names
-   * @return List of Pinot realtime table names
+   * Get all realtime table names.
+   *
+   * @return List of realtime table names
    */
-  @Nonnull
-  public List<String> getAllOfflineTables() {
-    List<String> resourceNames = getAllResources();
-    Iterator<String> iterator = resourceNames.iterator();
-    while (iterator.hasNext()) {
-      if (!TableNameBuilder.OFFLINE.tableHasTypeSuffix(iterator.next())) {
-        iterator.remove();
+  public List<String> getAllRealtimeTables() {
+    List<String> realtimeTableNames = new ArrayList<>();
+    for (String resourceName : getAllResources()) {
+      if (TableNameBuilder.REALTIME.tableHasTypeSuffix(resourceName)) {
```

Comment:
Same here. `TableNameBuilder.isRealtimeTableResource(resourceName)` might be better.